src/hotspot/share/memory/metaspace.cpp

```diff
@@ -569,12 +569,6 @@ void Metaspace::initialize_class_space(ReservedSpace rs) {
          "wrong alignment");
 
   MetaspaceContext::initialize_class_space_context(rs);
-
-  // This does currently not work because rs may be the result of a split
-  // operation and NMT seems not to be able to handle splits.
-  // Will be fixed with JDK-8243535.
-  // MemTracker::record_virtual_memory_type((address)rs.base(), mtClass);
```

Comment:
So why not just uncomment this line, instead of moving this to the caller?

Reply:
There are two initialization paths, one is used when running with CDS, and metaspace is initialized together with it, one is when running without CDS. This function is used for both. The CDS path is already covered though, I only need to do the NMT registration for the non-CDS case. Otherwise we register things doubly.

Reply:
Should have written that in the PR description, sorry.